### PR TITLE
Add metric for number of logins

### PR DESCRIPTION
**Pull request by tznamena**
Added `userlogins` count to the metrics collected by the deployment driver.
Refactored the login handler to allow calling the telemetry code after successful login.

### CODE REVIEW

File: server/engine/telemetry.go [hunk 1]
```diff
@@ -117,6 +118,22 @@ func SetFinalMetrics(db *gorm.DB) {
 
 }
 
+func UpdateLogInMetrics(db *gorm.DB) {
+	loginsCount := "1" // default value for case if no logins were recorded yet
```
Comment (bhavenst) [inline server/engine/telemetry.go:122]: Why not zero?

File: server/handler/login.go [hunk 2]
```diff
@@ -32,9 +40,11 @@ func GetLoginHandler(userName, userPassword string) http.HandlerFunc {
 				respondError(w, http.StatusInternalServerError, "Could not setup session")
 			}
 			respondJSON(w, http.StatusOK, map[string]string{"status": "success"})
+			// update metrics because user attempted logging in with correct credentials
+			engine.UpdateLogInMetrics(db)
```
Comment (bhavenst) [inline server/handler/login.go:44]: Tomorrow they will ask for failed login attempt count. :)

File: server/handler/login.go [hunk 3]
```diff
@@ -3,15 +3,23 @@ package handler
...
 type loginData struct {
 	Username string `json:"uid"`
 	Password string `json:"pwd"`
 }
 
-func GetLoginHandler(userName, userPassword string) http.HandlerFunc {
-	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+func GetLoginHandler() HandleFuncWithDB {
+	// initialization of the expected credentials
+	userName := "admin"
```
Comment (bhavenst) [inline server/handler/login.go:19]: Make a const instead?

**Review by bhavenst — APPROVED** [review]
Nothing major

File: server/engine/telemetry.go (hunk 1 above)
Comment (tznamena) [reply server/engine/telemetry.go:122]: This function is called only after successful login so this default of 1 means we don't have to do a +1 operation when no previous logins were recorded. If there were previous logins, the +1 is done right there (line 130)

File: server/handler/login.go (hunk 3 above)
Comment (tznamena) [reply server/handler/login.go:19]: Yup, will update it.

Comment (tznamena) [reply server/handler/login.go]: Done.

**Pull request merged**